src/PrestaShopBundle/Model/Product/AdminModelAdapter.php

```diff
@@ -532,9 +532,17 @@ function ($p) {
      */
     private function mapStep2FormData(Product $product)
     {
+        // ecotax is stored with tax included but form uses the tax excluded value
+        // using a precision of 6 digits as `AdminProductsController::_removeTaxFromEcotax()`
+        // which does the opposite uses 6 digits too
+        $ecotax = $this->tools->round(
+            $product->ecotax * (1 + $this->taxRuleDataProvider->getProductEcotaxRate() / 100),
+            6
```

Comment:
can we put it in a constant ?

Reply:
We could, but it would only be useful if all the places using this value use the constant as well.
I know https://github.com/PrestaShop/PrestaShop/blob/develop/controllers/admin/AdminProductsController.php#L2209 uses it as this function reverts the logic there, and there might be more.

Could it be considered an improvement and I'll do it in another PR ?
I'm afraid of changing other parts of the code and introducing crazy bugs 😅which would then be reported by QA and that would block this PR

Reply:
Ok with that